flink-java/src/main/java/org/apache/flink/api/java/operators/PartitionOperator.java

```diff
@@ -98,6 +101,14 @@ public PartitionOperator(DataSet<T> input, Keys<T> pKeys, Partitioner<?> customP
 		this.customPartitioner = customPartitioner;
 		this.distribution = distribution;
 	}
+
+	public PartitionOperator<T> withOrders(Order... orders) {
```

Comment:
The output of `ek.getOriginalKeyFieldTypes` should be: 
`TypeInformation[] {TypeInformation<Pojo1>, TypeInformation<Integer>}`.

So the composite type `Pojo1` should not be flattened into its components, i.e., `(String, String)`. This is important, because a user would only specify two orders (one for each key expression) and the first one needs to applied to all flat keys of `Pojo1`.